### PR TITLE
[Enhancment] Add Status::update()

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [] others

### CODE REVIEW

**Review by decster — DISMISSED** [review]

**Review by imay — DISMISSED** [review]

**Review by kevincai — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by sduzh** [comment]
run starrocks_be_build

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Comment by sduzh** [comment]
run starrocks_admit_test

**Comment by sduzh** [comment]
run starrocks_be_build

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Review by imay — APPROVED** [review]

File: be/src/common/status.h [hunk 1]
```diff
@@ -58,6 +58,25 @@ class Status {
...
+    // effect and preserves the current data.
+    //
+    // `update()` provides a convenient way of keeping track of the first error
+    // encountered.
+    //
+    // Example:
+    //   // Instead of "if (overall_status.ok()) overall_status = new_status"
+    //   overall_status.update(new_status);
+    //
+    void update(const Status& new_status);
```
Comment (mofeiatwork) [inline be/src/common/status.h:77]: why not naming it `update_if_ok`?

**Pull request merged**